nntrainer/layers/conv2d_layer.cpp

```diff
@@ -41,23 +43,18 @@ int Conv2DLayer::initialize() {
     ml_logw("Warning: the length of previous layer dimension is one");
   }
 
-  TensorDim dim =
-    TensorDim(1, in_dim.channel(), kernel_size[0], kernel_size[1]);
-  TensorDim bias_dim = TensorDim(1, 1, 1, 1);
-
   std::string kernelPrefix = "Conv2d:filter";
   std::string biasPrefix = "Conv2d:bias";
-  setNumWeights(filter_size * 2);
-
-  for (unsigned int i = 0; i < filter_size; ++i) {
-    /*< @note: order of weight and bias are:
-               w0 w1 w2 ... w(filter_size) b0 b1 b2 ... b(filter_size)
-    */
-    weightAt(i) = std::move(
-      Weight(dim, weight_initializer, true, kernelPrefix + std::to_string(i)));
-    weightAt(i + filter_size) = std::move(
-      Weight(bias_dim, bias_initializer, true, biasPrefix + std::to_string(i)));
-  }
+
+  TensorDim dim =
+    TensorDim(filter_size, in_dim.channel(), kernel_size[0], kernel_size[1]);
+  TensorDim bias_dim = TensorDim(1, filter_size, 1, 1);
+  setNumWeights(2);
+
+  weightAt(ConvParams::weight) =
+    std::move(Weight(dim, weight_initializer, true, kernelPrefix));
```

Comment:
Not related to the PR though, right side is already rvalue so std::move is redundant.

Reply:
good catch, removed.